AirCastingTests/Regressions/SDSync/SDSyncRegressions.swift

```diff
@@ -0,0 +1,130 @@
+// Created by Lunar on 12/07/2022.
+//
+
+import XCTest
+import CoreBluetooth
+import Resolver
+import Combine
+import CoreData
+@testable import AirCasting
+
+// swiftlint:disable print_using
+
+class SDSyncRegressions: ACTestCase {
+    let uuid = UUID()
+    let measurementsCount = 100_000
+    lazy var airBeamServices = AirBeamServices(measurementCount: measurementsCount, uuid: uuid)
+    lazy var uploadServices = UploadFixedSessionService()
+    lazy var fixedServices = FixedSessionUpdatingService()
+    lazy var synchronizer = Synchronizer()
+    lazy var persistence = PersistenceController(inMemory: true)
+    
+    override func setUp() {
+        // Clear Documents and Temp directory
+        let documentsURL = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first!
+        let tempURL = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first!
+        try! FileManager.default.clearDirectory(at: documentsURL)
+        try! FileManager.default.clearDirectory(at: tempURL)
+        
+        super.setUp()
+        
+        Resolver.test.register { self.airBeamServices as SDCardAirBeamServices }
+        Resolver.test.register { self.uploadServices as UploadFixedSessionAPIService }
+        Resolver.test.register { self.fixedServices as MeasurementUpdatingService }
+        Resolver.test.register { self.persistence as PersistenceController }
+    }
+    
+    //
+    // Regression source:
+    // https://trello.com/c/VeJ8oOaL
+    // SD sync fails when SD card contains lots of data
+    //
+    func test_whenAirbeamHasLotsOfMeasurementsToSync_theAppCrashes() throws {
```

Comment:
this only worked once for me and it's sitting in the same target as our other tests, so it will delay tests execution by 1m40s... Not sure what should we do with that, but it shouldn't go into development branch. Maybe comment out?